src/cirrus/lib/process_payload.py

```diff
@@ -277,14 +333,58 @@ def payload_ids(self) -> list[str]:
         """
         return [c["id"] for c in self.payloads]
 
-    def get_states(self) -> dict[str, StateEnum]:
+    def get_states_and_exec_arn(self) -> dict[str, tuple]:
         if self.state_items is None:
             items = [
                 self.statedb.dbitem_to_item(i)
                 for i in self.statedb.get_dbitems(self.payload_ids)
             ]
             self.state_items = items
```

Comment:
Not that this is a suggestion to change anything, but it popped out at me because I recent learned about [the `cached_property` decorator](https://docs.python.org/3/library/functools.html#functools.cached_property). We could leverage it here like so:

```python
    @cached_property
    def state_items(self) -> SOME_TYPE_IM_TOO_LAZY_TO_LOOK_UP:
        return [
            self.statedb.dbitem_to_item(i)
            for i in self.statedb.get_dbitems(self.payload_ids)
        ]

    def get_states_and_exec_arn(self) -> dict[str, tuple]:
        response = dict(self.get_process_attrs(self.state_items))
        ...
```

Actually, now that I look at this more, it's kinda funny that we have `state_items` on `self` but no way to set it _without_ calling this method. That weirdness seems like something we should resolve, but I still won't ask for that to happen as part of this work unless my pointing it out makes it a compelling thing to fix.